### ISSUE TITLE
1.5.4 Crashes on RHEL 7.6 / Linux Mint 17 Long term support Current till April 2018

### ISSUE THREAD

**Issue by magnum93** [Foundry376/Mailspring#1292]
Hi, Tried to update to 1.5.4 version latest I believe on Linux mint 17 Qiana 64 bit.
Application crashes to console mode / developer mode showing padding.

**Comment by bengotow** [comment]
Hey! Could you take a screenshot and post it here? We had some reports of issues with RHEL 7.6 as well and I'm hoping it's the same error!

**Comment by rschroll** [comment]
I'm seeing the same behavior on Ubuntu 14.04, installed from the 64-bit deb.  Launching Mailspring from the command line yields this output:

```
mailspring: /lib/x86_64-linux-gnu/libdbus-1.so.3: no version information available (required by mailspring)
/usr/share/mailspring/mailspring: /lib/x86_64-linux-gnu/libdbus-1.so.3: no version information available (required by /usr/share/mailspring/mailspring)
Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
Running database migrations
App load time: 210ms
/usr/share/mailspring/resources/app.asar.unpacked/mailsync: line 4: realpath: command not found
dirname: missing operand
Try 'dirname --help' for more information.
/usr/share/mailspring/resources/app.asar.unpacked/mailsync: line 4: realpath: command not found
dirname: missing operand
Try 'dirname --help' for more information.

{"error":null}
Manual update check (updates.getmailspring.com/check/linux/x64/1.5.4-1d923cf4/d24bde72-a836-47dc-8359-60c87a68866c/stable) returned 204
{ Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/better-sqlite3/build/Debug/better_sqlite3.node)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at bindings (/usr/share/mailspring/resources/app.asar/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/lib/database.js:5:38)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/lib/database.js:50:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/index.js:2:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/index.js:36:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/src/flux/stores/database-store.es6:6:1)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/src/flux/stores/database-store.js:521:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/src/flux/stores/task-queue.es6:5:1)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/src/flux/stores/task-queue.js:160:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at lazyLoadWithGetter (/src/global/mailspring-exports.es6:31:48)
    at Object.get [as TaskQueue] (/src/global/mailspring-exports.es6:23:21)
    at process.nextTick (/src/global/mailspring-exports.es6:205:10)
    at process._tickCallback (internal/process/next_tick.js:61:11)
  message:
   '/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9\' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/better-sqlite3/build/Debug/better_sqlite3.node)' } { url:
   '/usr/share/mailspring/resources/app.asar/node_modules/bindings/bindings.js',
  line: 88,
  column: 9,
  pluginIds: [] }
{ Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/better-sqlite3/build/Debug/better_sqlite3.node)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at bindings (/usr/share/mailspring/resources/app.asar/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/lib/database.js:5:38)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/lib/database.js:50:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/index.js:2:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/better-sqlite3/index.js:36:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/src/flux/stores/database-store.es6:6:1)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/src/flux/stores/database-store.js:521:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/src/flux/stores/task-queue.es6:5:1)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/src/flux/stores/task-queue.js:160:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at lazyLoadWithGetter (/src/global/mailspring-exports.es6:31:48)
    at Object.get [as TaskQueue] (/src/global/mailspring-exports.es6:23:21)
    at process.nextTick (/src/global/mailspring-exports.es6:205:10)
    at process._tickCallback (internal/process/next_tick.js:61:11)
  message:
   '/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9\' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/better-sqlite3/build/Debug/better_sqlite3.node)' } { url:
   '/usr/share/mailspring/resources/app.asar/node_modules/bindings/bindings.js',
  line: 88,
  column: 9,
  pluginIds: [] }
{ Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/@paulcbetts/spellchecker/build/Release/spellchecker.node)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/@paulcbetts/spellchecker/lib/spellchecker.js:3:16)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/@paulcbetts/spellchecker/lib/spellchecker.js:116:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/node-spellchecker.js:8:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/node-spellchecker.js:9:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/index.js:19:25)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/index.js:46:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at initHandler (/src/spellchecker.es6:26:37)
  message:
   '/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9\' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/@paulcbetts/spellchecker/build/Release/spellchecker.node)' } { url: 'ELECTRON_ASAR.js', line: 160, column: 31, pluginIds: [] }
{ Error: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/@paulcbetts/spellchecker/build/Release/spellchecker.node)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:160:31)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:722:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:169:18)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/@paulcbetts/spellchecker/lib/spellchecker.js:3:16)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/@paulcbetts/spellchecker/lib/spellchecker.js:116:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/node-spellchecker.js:8:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/node-spellchecker.js:9:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/index.js:19:25)
    at Object.<anonymous> (/usr/share/mailspring/resources/app.asar/node_modules/electron-spellchecker/lib/index.js:46:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
    at Function.Module._load (internal/modules/cjs/loader.js:533:3)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at initHandler (/src/spellchecker.es6:26:37)
  message:
   '/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9\' not found (required by /usr/share/mailspring/resources/app.asar.unpacked/node_modules/@paulcbetts/spellchecker/build/Release/spellchecker.node)' } { url: 'ELECTRON_ASAR.js', line: 160, column: 31, pluginIds: [] }
```

**Comment by rschroll** [comment]
FWIW, I have `libstdc++` 4.8.4; Google suggests that `CXXABI_1.3.9` requires 5.1.

**Comment by bengotow** [comment]
Hey folks—thanks for reporting this. I don't know much about this but I'm doing my best to track down what changed. Aside from upgrading Electron to v4 (which /shouldn't/ have caused it to use a newer C++ compiler or libstdc, as far as I know) the only thing that changed is that Travis upgraded a few things. I've reverted teh Electron 4 upgrade on a branch and I'm running a CI build to determine if this is caused by a CI environment change or by the upgrade. Should have more info soon.

**Comment by bengotow** [comment]
Confirmed via our support emails that this is also impacting Red Hat Enterprise Linux 7.6

**Comment by bengotow** [comment]
This was fixed in the 1.5.5 release - going to go ahead and close this!
